server/devdocs/index.js

```diff
@@ -196,5 +197,20 @@ module.exports = function() {
 		response.send( marked( fileContents ) );
 	} );
 
+	// return json for the components usage stats
+	app.get( '/devdocs/service/usage-stats', function( request, response ) {
+		var counts = Object.keys( usageStats )
+			.filter( function( moduleName ) {
+				return moduleName.indexOf( 'components/' ) === 0 &&
```

Comment:
Oh, that would mean moving the logic on the client-side. I meant just moving the logic itself out of the controller, probably into a function.